foxpuppet/windows/browser/bookmarks/bookmark.py

```
@@ -0,0 +1,212 @@
+                bookmark_menu.click()
+                menu_items = bookmark_menu.find_elements(
+                    By.CSS_SELECTOR, "menuitem.bookmark-item"
+                )
+                for item in menu_items:
+                    item_label = item.get_attribute("label")
+                    if item_label and item_label.lower() in current_page_title.lower():
+                        return True
+        return False
+
+    def add(self) -> None:
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.selenium.find_element(*BookmarkLocators.STAR_BUTTON).click()
+            self.selenium.find_element(*BookmarkLocators.FOLDER_MENU).click()
+            self.selenium.find_element(*BookmarkLocators.OTHER_BOOKMARKS_STAR).click()
+            self.selenium.find_element(*BookmarkLocators.SAVE_BUTTON).click()
+
+    def add_bookmark(self, bookmark_data: BookmarkData) -> None:
+        """Add a Bookmark using the main bookmark menu."""
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.actions.context_click(
+                self.selenium.find_element(*BookmarkLocators.NAVIGATOR_TOOLBOX)
+            ).perform()
+            self.selenium.find_element(*BookmarkLocators.MENU_BAR).click()
+            self.selenium.find_element(*BookmarkLocators.MAIN_MENU_BOOKMARK).click()
+            self.actions.context_click(
+                self.selenium.find_element(*BookmarkLocators.MANAGE_BOOKMARKS)
+            ).perform()
+            self.selenium.find_element(*BookmarkLocators.ADD_BOOKMARK).click()
+
+            frame_element = self.selenium.find_element(
+                *BookmarkLocators.ADD_BOOKMARK_FRAME
+            )
+            if frame_element is not None:
+                self.selenium.switch_to.frame(frame_element)
+            else:
+                raise NoSuchElementException(
+                    f"iFrame with locator {BookmarkLocators.ADD_BOOKMARK_FRAME} not found."
+                )
```

Selenium will raise an error before the program gets here, if you want to raise an error with specific context you'll have to use a `try/except` block.